### PR TITLE
feat: Add GetProductVersionFeatureFlag function

### PR DESCRIPTION
**Pull request by muneebkq**
This commit adds the GetProductVersionFeatureFlag function to the LexActivator module. This function allows retrieving the feature flag status for a specific product version. It takes the name of the feature flag, a pointer to store the enabled status, a pointer to store the data associated with the feature flag, and the length of the data.

### CODE REVIEW

File: examples/LexActivator.rb [hunk 1]
```diff
@@ -112,6 +112,15 @@ module PermissionFlags
   # @scope class
   attach_function :GetProductMetadata, :GetProductMetadata, [:string, :pointer, :uint], :int
 
+  # @method GetProductVersionFeatureFlag(name, enabled, data, length)
+  # @param [String] name
+  # @param [String] enabled
```
Comment (ahmad-kemsan) [inline examples/LexActivator.rb]: @muneebkq enabled is not a string, it should be a boolean.

**Review by ahmad-kemsan — CHANGES_REQUESTED** [review]
Please check the same function in other wrappers

File: examples/LexActivator.rb (hunk 1 above)
Comment (muneebkq) [reply examples/LexActivator.rb]: refactor done

File: examples/LexStatusCodes.rb [hunk 2]
```diff
@@ -42,6 +42,11 @@ module LexStatusCodes
 
   LA_RELEASE_NO_UPDATE_AVAILABLE = 31
 
+  # CODE: LA_RELEASE_UPDATE_AVAILABLE_NOT_ALLOWED
+  # MESSAGE: The update available is not allowed for this license.
+
+  LA_RELEASE_UPDATE_AVAILABLE_NOT_ALLOWED = 32,
```
Comment (ahmad-kemsan) [inline examples/LexStatusCodes.rb]: @muneebkq This error code would be possible via CheckReleaseUpdate() only and that is not included yet.

File: examples/LexStatusCodes.rb [hunk 3]
```diff
@@ -195,6 +224,34 @@ module LexStatusCodes
...
+  # MESSAGE: The user is not authenticated.
+  LA_E_USER_NOT_AUTHENTICATED = 87,
+
+  # CODE: LA_E_TWO_FACTOR_AUTHENTICATION_CODE_MISSING
+  # MESSAGE: The two-factor authentication code for the user authentication is missing.
+  LA_E_TWO_FACTOR_AUTHENTICATION_CODE_MISSING = 88,
+
+  # CODE: LA_E_TWO_FACTOR_AUTHENTICATION_CODE_INVALID
+  # MESSAGE: The two-factor authentication code provided by the user is invalid.
+  LA_E_TWO_FACTOR_AUTHENTICATION_CODE_INVALID = 89,
```
Comment (ahmad-kemsan) [inline examples/LexStatusCodes.rb]: @muneebkq What about the error codes from 100 -105? They seem to be not included.

File: examples/LexActivator.rb [hunk 4]
```diff
@@ -112,6 +223,59 @@ module PermissionFlags
...
+
+  # @method GetLicenseCreationDate(creationDate)
+  # @param [FFI::Pointer(*Uint32T)] creationDate
+  # @return [Integer]
+  # @scope class
+  attach_function :GetLicenseCreationDate, :GetLicenseCreationDate, [:pointer], :int
+
+  # @method GetProductVersionFeatureFlag(name, enabled, data, length)
+  # @param [String] name
+  # @param [Integer] enabled
```
Comment (ahmad-kemsan) [inline examples/LexActivator.rb]: @muneebkq param docs for enabled would be helpful.

File: examples/LexActivator.rb [hunk 5]
```diff
@@ -31,9 +31,59 @@ def self.decode_utf16(input)
...
+
+  class OrganizationAddress < FFI::Struct
+    layout :addressLine1, [:char, BUFFER_SIZE_256],
+           :addressLine2, [:char, BUFFER_SIZE_256],
+           :city, [:char, BUFFER_SIZE_256],
+           :state, [:char, BUFFER_SIZE_256],
+           :country, [:char, BUFFER_SIZE_256],
+           :postalCode, [:char, BUFFER_SIZE_256]
+
+    def to_string
```
Comment (ahmad-kemsan) [inline examples/LexActivator.rb]: @muneebkq `to_string` is not a very idiomatic name in Ruby, and that `to_hash` would be a more appropriate and descriptive name for a method that converts an FFI::Struct to a Hash.

File: examples/LexActivator.rb [hunk 6]
```diff
@@ -31,9 +31,59 @@ def self.decode_utf16(input)
...
+
+  class UserLicense < FFI::Struct
+    layout :allowedActivations, :int64,
+           :allowedDeactivations, :int64,
+           :key, [:char, BUFFER_SIZE_256],
+           :type, [:char, BUFFER_SIZE_256],
+           :metadata, [Metadata, MAX_METADATA_SIZE] 
+
+    
+    def to_string
```
Comment (ahmad-kemsan) [inline examples/LexActivator.rb]: @muneebkq Same here for the method name

File: examples/LexActivator.rb [hunk 7]
```diff
@@ -170,13 +348,40 @@ module PermissionFlags
   # @scope class
   attach_function :GetLicenseUserMetadata, :GetLicenseUserMetadata, [:string, :pointer, :uint], :int
 
+  # @method GetLicenseOrganizationAddress(organizationAddress)
+  # @param [String] organizationAddress
```
Comment (ahmad-kemsan) [inline examples/LexActivator.rb]: @muneebkq the param is a pointer to the struct. Please update the param docs.

File: examples/LexActivator.rb [hunk 8]
```diff
@@ -170,13 +348,40 @@ module PermissionFlags
...
+  # @return [Integer]
+  # @scope class
+  attach_function :GetLicenseOrganizationAddress, [:pointer], :int
+
+  # @method GetUserLicenses(UserLicense, length)
+  # @param [String] UserLicense
```
Comment (ahmad-kemsan) [inline examples/LexActivator.rb]: @muneebkq Same here param is a pointer to the struct.

**Review by ahmad-kemsan — CHANGES_REQUESTED** [review]
Please check the comments

**Review by ahmad-kemsan — APPROVED** [review]

**Pull request merged**